### ISSUE TITLE
Unable to find package `salt` from 2016.11 repo in Cloudbreak CentOS 7

### ISSUE THREAD

**Issue by mtslzr** [saltstack/salt#56442]
### Description of Issue
I know this might be a weird, niche problem, but I'm trying to set up Cloudera Cloudbreak CentOS 7 on AWS (AMI IDs can be found here: https://cloudbreak-imagecatalog.s3.amazonaws.com/v3-rc-cb-image-catalog.json; base is CentOS 7).

Can install the 2016.11 EL7 repo with `yum -y install https://repo.saltstack.com/yum/redhat/salt-repo-2016.11-2.el7.noarch.rpm` no problem. But the repo then doesn't show the `salt` package. All the others show up:

```
# yum search salt
Loaded plugins: fastestmirror, versionlock
=============================================================================== N/S matched: salt ===============================================================================
perl-Crypt-SaltedHash.noarch : Perl interface to functions that assist in working with salted hashes
python-salttesting.noarch : Testing library for SaltStack projects
salt-api.noarch : REST API for Salt, a parallel remote execution system
salt-cloud.noarch : Cloud provisioner for Salt, a parallel remote execution system
salt-master.noarch : Management component for salt, a parallel remote execution system
salt-minion.noarch : Client component for Salt, a parallel remote execution system
salt-ssh.noarch : Agentless SSH-based version of Salt, a parallel remote execution system
salt-syndic.noarch : Master-of-master component for Salt, a parallel remote execution system
salt-repo.noarch : Repo configuration for 2016.11 release cycle
```

Setting up a vanilla CentOS 7 instance with the same process shows the package. Likewise, manually pulling the RPM from the web with `wget` and manually installing dependencies works around it, but isn't ideal for this setup.

### Setup
Configure an EC2 instance with Cloudera's Cloudbreak CentOS 7 AMI (`ami-0f51ac4606a1ced39`). 

### Steps to Reproduce Issue
Pull down the 2016.11 EL7 repo (yum -y install https://repo.saltstack.com/yum/redhat/salt-repo-2016.11-2.el7.noarch.rpm`). Attempt to install salt-minion with `yum -y install salt-minion`. It will request `salt` as a dependency and bomb out. Can also run `yum search salt` or `yum info salt` to see the missing package.

### Versions Report
Salt 2016.11
`uname -r | 3.10.0-1062.12.1.el7.x86_64`
`cat /etc/redhat-release | CentOS Linux release 7.7.1908 (Core)`

I know this is a longshot, but not sure how else to proceed and niche-ness aside, seems weird that all but one package show up normally in the repo in this case.

**Comment by dmurphy18** [comment]
@mtslzr Is there a reason for picking 2016.11, which is a very old branch and has long been end of life'd, 2019.2.3 would be a better suggestion to use.

What is the result of yum repolist and can you display the contents of /etc/yum.repo.d/salt-xxxxxx.repo

**Comment by mtslzr** [comment]
@dmurphy18 It's not ideal, but company currently requires it (believe it's tied to Python 2.7; there's a big upgrade scheduled, but for the moment I'm stuck with 2016.11).

```
# yum repolist
Loaded plugins: fastestmirror, versionlock
Loading mirror speeds from cached hostfile
 * base: d36uatko69830t.cloudfront.net
 * epel: iad.mirror.rackspace.com
 * extras: d36uatko69830t.cloudfront.net
 * updates: d36uatko69830t.cloudfront.net
repo id                                                              repo name                                                                                           status
base/7/x86_64                                                        CentOS-7 - Base                                                                                       10,097
epel/x86_64                                                          Extra Packages for Enterprise Linux 7 - x86_64                                                      13,216+1
extras/7/x86_64                                                      CentOS-7 - Extras                                                                                        335
pgdg96                                                               PostgreSQL 9.6                                                                                         1,077
salt-2016.11/x86_64                                                  SaltStack 2016.11 Release Channel for RHEL/Centos 7                                                    75+26
treasuredata/7/x86_64                                                TreasureData                                                                                              13
updates/7/x86_64                                                     CentOS-7 - Updates                                                                                     1,774
repolist: 26,587
```

```
[salt-2016.11]
name=SaltStack 2016.11 Release Channel for RHEL/Centos $releasever
baseurl=https://repo.saltstack.com/yum/redhat/7/$basearch/2016.11
failovermethod=priority
enabled=1
gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/saltstack-signing-key
```

Will try to build a work-around (I know I'm casting a real, real tiny net here) here, but figured it couldn't hurt to mention it (in case it ever impacts other OSes/versions potentially).

**Comment by mtslzr** [comment]
For what it's worth, the issue is identical with the latest Salt repo. Sees every other package, but not `salt`.

**Comment by dmurphy18** [comment]
Have to believe you are doing something wrong or your environment is messed up, otherwise a lot more people would be complaining about it, that and the fact I unit test these before they ever get close to QA, who do more.  Installing on a Redhat 7.7 and found no issue

```
root@Unknown:~# yum repolist
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
 * base: mirrors.unifiedlayer.com
 * epel: mirror.chpc.utah.edu
 * extras: mirrors.unifiedlayer.com
 * updates: mirrors.unifiedlayer.com
repo id                                                                                               repo name                                                                                                                          status
!base/7/x86_64                                                                                        CentOS-7 - Base                                                                                                                    10,097
!epel/x86_64                                                                                          Extra Packages for Enterprise Linux 7 - x86_64                                                                                     13,217
!extras/7/x86_64                                                                                      CentOS-7 - Extras                                                                                                                     335
!updates/7/x86_64                                                                                     CentOS-7 - Updates                                                                                                                  1,774
repolist: 25,423
root@Unknown:~# yum install https://repo.saltstack.com/yum/redhat/salt-repo-2016.11-2.el7.noarch.rpm 
Loaded plugins: fastestmirror, langpacks
salt-repo-2016.11-2.el7.noarch.rpm                                                                                                                                                                                      | 5.6 kB  00:00:00     
Examining /var/tmp/yum-root-X8TUrz/salt-repo-2016.11-2.el7.noarch.rpm: salt-repo-2016.11-2.el7.noarch
Marking /var/tmp/yum-root-X8TUrz/salt-repo-2016.11-2.el7.noarch.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package salt-repo.noarch 0:2016.11-2.el7 will be installed
--> Finished Dependency Resolution
base/7/x86_64                                                                                                                                                                                                           | 3.6 kB  00:00:00     
epel/x86_64/metalink                                                                                                                                                                                                    |  18 kB  00:00:00     
epel/x86_64                                                                                                                                                                                                             | 4.7 kB  00:00:00     
epel/x86_64/updateinfo                                                                                                                                                                                                  | 1.0 MB  00:00:00     
epel/x86_64/primary_db                                                                                                                                                                                                  | 6.7 MB  00:00:01     
extras/7/x86_64                                                                                                                                                                                                         | 2.9 kB  00:00:00     
updates/7/x86_64                                                                                                                                                                                                        | 2.9 kB  00:00:00     

Dependencies Resolved

===============================================================================================================================================================================================================================================
 Package                                             Arch                                             Version                                                  Repository                                                                 Size
===============================================================================================================================================================================================================================================
Installing:
 salt-repo                                           noarch                                           2016.11-2.el7                                            /salt-repo-2016.11-2.el7.noarch                                           3.6 k

Transaction Summary
===============================================================================================================================================================================================================================================
Install  1 Package

Total size: 3.6 k
Installed size: 3.6 k
Is this ok [y/d/N]: y
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : salt-repo-2016.11-2.el7.noarch                                                                                                                                                                                              1/1 
  Verifying  : salt-repo-2016.11-2.el7.noarch                                                                                                                                                                                              1/1 

Installed:
  salt-repo.noarch 0:2016.11-2.el7                                                                                                                                                                                                             

Complete!
root@Unknown:~# yum list salt-minion
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
 * base: mirrors.unifiedlayer.com
 * epel: mirror.chpc.utah.edu
 * extras: mirrors.unifiedlayer.com
 * updates: mirrors.unifiedlayer.com
salt-2016.11                                                                                                                                                                                                            | 2.9 kB  00:00:00     
salt-2016.11/x86_64/primary_db                                                                                                                                                                                          |  37 kB  00:00:00     
Available Packages
salt-minion.noarch                    
root@Unknown:~# yum erase salt-bin-arista
Loaded plugins: fastestmirror, langpacks
Resolving Dependencies
--> Running transaction check
---> Package salt-bin-arista.x86_64 0:3000-4.el7 will be erased
--> Finished Dependency Resolution

Dependencies Resolved

===============================================================================================================================================================================================================================================
 Package                                                 Arch                                           Version                                              Repository                                                                   Size
===============================================================================================================================================================================================================================================
Removing:
 salt-bin-arista                                         x86_64                                         3000-4.el7                                           @/salt-bin-arista-3000-4.el7.x86_64                                         125 M

Transaction Summary
===============================================================================================================================================================================================================================================
Remove  1 Package

Installed size: 125 M
Is this ok [y/N]: y
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Erasing    : salt-bin-arista-3000-4.el7.x86_64                                                                                                                                                                                           1/1 
warning: /etc/salt/minion saved as /etc/salt/minion.rpmsave
  Verifying  : salt-bin-arista-3000-4.el7.x86_64                                                                                                                                                                                           1/1 

Removed:
  salt-bin-arista.x86_64 0:3000-4.el7                                                                                                                                                                                                          

Complete!
root@Unknown:~# yum install salt-minion
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
 * base: mirrors.unifiedlayer.com
 * epel: mirror.chpc.utah.edu
 * extras: mirrors.unifiedlayer.com
 * updates: mirrors.unifiedlayer.com
Resolving Dependencies
--> Running transaction check
---> Package salt-minion.noarch 0:2016.11.10-1.el7 will be installed
--> Processing Dependency: salt = 2016.11.10-1.el7 for package: salt-minion-2016.11.10-1.el7.noarch
--> Running transaction check
---> Package salt.noarch 0:2016.11.10-1.el7 will be installed
--> Processing Dependency: python-msgpack > 0.3 for package: salt-2016.11.10-1.el7.noarch
--> Processing Dependency: python-zmq for package: salt-2016.11.10-1.el7.noarch
--> Running transaction check
---> Package python-zmq.x86_64 0:15.3.0-3.el7 will be installed
--> Processing Dependency: zeromq >= 4 for package: python-zmq-15.3.0-3.el7.x86_64
--> Processing Dependency: libzmq.so.5()(64bit) for package: python-zmq-15.3.0-3.el7.x86_64
---> Package python2-msgpack.x86_64 0:0.5.6-5.el7 will be installed
--> Running transaction check
---> Package zeromq.x86_64 0:4.1.4-7.el7 will be installed
--> Processing Dependency: libsodium.so.23()(64bit) for package: zeromq-4.1.4-7.el7.x86_64
--> Processing Dependency: libpgm-5.2.so.0()(64bit) for package: zeromq-4.1.4-7.el7.x86_64
--> Running transaction check
---> Package libsodium.x86_64 0:1.0.18-1.el7 will be installed
---> Package openpgm.x86_64 0:5.2.122-2.el7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

===============================================================================================================================================================================================================================================
 Package                                                     Arch                                               Version                                                         Repository                                                Size
===============================================================================================================================================================================================================================================
Installing:
 salt-minion                                                 noarch                                             2016.11.10-1.el7                                                salt-2016.11                                              35 k
Installing for dependencies:
 libsodium                                                   x86_64                                             1.0.18-1.el7                                                    epel                                                     147 k
 openpgm                                                     x86_64                                             5.2.122-2.el7                                                   epel                                                     171 k
 python-zmq                                                  x86_64                                             15.3.0-3.el7                                                    salt-2016.11                                             520 k
 python2-msgpack                                             x86_64                                             0.5.6-5.el7                                                     epel                                                      64 k
 salt                                                        noarch                                             2016.11.10-1.el7                                                salt-2016.11                                             7.1 M
 zeromq                                                      x86_64                                             4.1.4-7.el7                                                     salt-2016.11                                             556 k

Transaction Summary
===============================================================================================================================================================================================================================================
Install  1 Package (+6 Dependent packages)

Total size: 8.6 M
Installed size: 37 M
Is this ok [y/d/N]: y
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : openpgm-5.2.122-2.el7.x86_64                                                                                                                                                                                                1/7 
  Installing : python2-msgpack-0.5.6-5.el7.x86_64                                                                                                                                                                                          2/7 
  Installing : libsodium-1.0.18-1.el7.x86_64                                                                                                                                                                                               3/7 
  Installing : zeromq-4.1.4-7.el7.x86_64                                                                                                                                                                                                   4/7 
  Installing : python-zmq-15.3.0-3.el7.x86_64                                                                                                                                                                                              5/7 
  Installing : salt-2016.11.10-1.el7.noarch                                                                                                                                                                                                6/7 
  Installing : salt-minion-2016.11.10-1.el7.noarch                                                                                                                                                                                         7/7 
  Verifying  : libsodium-1.0.18-1.el7.x86_64                                                                                                                                                                                               1/7 
  Verifying  : zeromq-4.1.4-7.el7.x86_64                                                                                                                                                                                                   2/7 
  Verifying  : python2-msgpack-0.5.6-5.el7.x86_64                                                                                                                                                                                          3/7 
  Verifying  : salt-minion-2016.11.10-1.el7.noarch                                                                                                                                                                                         4/7 
  Verifying  : openpgm-5.2.122-2.el7.x86_64                                                                                                                                                                                                5/7 
  Verifying  : salt-2016.11.10-1.el7.noarch                                                                                                                                                                                                6/7 
  Verifying  : python-zmq-15.3.0-3.el7.x86_64                                                                                                                                                                                              7/7 

Installed:
  salt-minion.noarch 0:2016.11.10-1.el7                                                                                                                                                                                                        

Dependency Installed:
  libsodium.x86_64 0:1.0.18-1.el7        openpgm.x86_64 0:5.2.122-2.el7        python-zmq.x86_64 0:15.3.0-3.el7        python2-msgpack.x86_64 0:0.5.6-5.el7        salt.noarch 0:2016.11.10-1.el7        zeromq.x86_64 0:4.1.4-7.el7       

Complete!
root@Unknown:~# 
```

**Comment by dmurphy18** [comment]
This was not Cloudera but an actual RHEL 7.7 installed in VirtualBox.  What is the output of 'yum install salt-minion', want to see the dependencies it thinks are needed, given salt and salt-minion are from the same repository, if you can access one, you should be able to access the other

**Comment by mtslzr** [comment]
```# yum install salt-minion
Loaded plugins: fastestmirror, versionlock
Loading mirror speeds from cached hostfile
 * base: d36uatko69830t.cloudfront.net
 * epel: iad.mirror.rackspace.com
 * extras: d36uatko69830t.cloudfront.net
 * updates: d36uatko69830t.cloudfront.net
Resolving Dependencies
--> Running transaction check
---> Package salt-minion.noarch 0:2016.11.10-1.el7 will be installed
--> Processing Dependency: salt = 2016.11.10-1.el7 for package: salt-minion-2016.11.10-1.el7.noarch
--> Finished Dependency Resolution
Error: Package: salt-minion-2016.11.10-1.el7.noarch (salt-2016.11)
           Requires: salt = 2016.11.10-1.el7
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest```

**Comment by dmurphy18** [comment]
What is the result of 'yum list salt*' on your machine

**Comment by mtslzr** [comment]
```
# yum list 'salt*'
Loaded plugins: fastestmirror, versionlock
Loading mirror speeds from cached hostfile
 * base: d36uatko69830t.cloudfront.net
 * epel: iad.mirror.rackspace.com
 * extras: d36uatko69830t.cloudfront.net
 * updates: d36uatko69830t.cloudfront.net
Installed Packages
salt-repo.noarch                                                         2016.11-2.el7                                                           @/salt-repo-2016.11-2.el7.noarch
Available Packages
salt-api.noarch                                                          2016.11.10-1.el7                                                        salt-2016.11
salt-cloud.noarch                                                        2016.11.10-1.el7                                                        salt-2016.11
salt-master.noarch                                                       2016.11.10-1.el7                                                        salt-2016.11
salt-minion.noarch                                                       2016.11.10-1.el7                                                        salt-2016.11
salt-ssh.noarch                                                          2016.11.10-1.el7                                                        salt-2016.11
salt-syndic.noarch                                                       2016.11.10-1.el7                                                        salt-2016.11
```

**Comment by dmurphy18** [comment]
Extremely odd that it can see the others but not salt, it is as if something is limiting files in your environment since from my machine at home outside of VPN's etc., salt is available from the repo
```
root@Unknown:~# yum list salt*
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
 * base: mirrors.unifiedlayer.com
 * epel: mirror.chpc.utah.edu
 * extras: mirrors.unifiedlayer.com
 * updates: mirrors.unifiedlayer.com
Installed Packages
salt.noarch                                                                                             2016.11.10-1.el7                                                                                       @salt-2016.11                   
salt-minion.noarch                                                                                      2016.11.10-1.el7                                                                                       @salt-2016.11                   
salt-repo.noarch                                                                                        2016.11-2.el7                                                                                          @/salt-repo-2016.11-2.el7.noarch
Available Packages
salt-api.noarch                                                                                         2016.11.10-1.el7                                                                                       salt-2016.11                    
salt-cloud.noarch                                                                                       2016.11.10-1.el7                                                                                       salt-2016.11                    
salt-master.noarch                                                                                      2016.11.10-1.el7                                                                                       salt-2016.11                    
salt-ssh.noarch                                                                                         2016.11.10-1.el7                                                                                       salt-2016.11                    
salt-syndic.noarch                                                                                      2016.11.10-1.el7                                                                                       salt-2016.11                    
root@Unknown:~# 
```

Perhaps you should check your environment settings to try and understandard why salt.noarch is not available, perhaps check from your web browser the following link to show that salt.noarch is in the repository  http://repo.saltstack.com/yum/redhat/7/x86_64/2016.11/  and http://repo.saltstack.com/yum/redhat/7/x86_64/2016.11/salt-2016.11.10-1.el7.noarch.rpm

**Comment by mtslzr** [comment]
The package exists via the web (using the links you posted); my current work-around is actually to pull it down via `wget` and install it locally. Which works. Have to manually install some dependencies, then the RPM, then I can `yum install salt-minion`.

Definitely super bizarre. Will keep digging, but figured it couldn't hurt to put it on the radar. Thanks for all the help.

**Comment by dmurphy18** [comment]
@mtslzr Wondering if you have made any progress as to why package 'salt' was not available.Several years ago, SaltStack had some scripts update some packages and for example the globing had an issue, for example: ** salt-* ** rather than ** salt* ** (it was actually something else than salt, but using this as an example).  Hence I am wondering if there is something similar in your environment that may be doing something similar.

**Comment by mtslzr** [comment]
Unfortunately have not yet. Ended up writing a work-around (pulling `salt` from wget, installing dependencies, then `yum install -y salt-minion`) for now. I know we're working on moving away from 2016.11 "soon", so not sure how much bandwidth I'll get to dig more into it. That said, let me do some testing around the globing you mentioned, and if anything comes up, I'll update the issue.

I'm guessing the Venn diagram of people using a Cloudbreak CentOS 7 AWS AMI and people still on Salt 2016.11 is basically me, heh. So hopefully this is just some oddity as a result of the super-custom AMI (FWIW, we require SELinux, which has gunked up a number of other bits of our build process with this image as well, so could be something with that causing a ruckus).

**Comment by mtslzr** [comment]
In the wake of numerous older versions (included 2016.11) being shelved, I'll close this. We ended up just using `pip install` to install Salt, and have moved away from 2016.11 now anyway. Thanks for the help.
